### PR TITLE
implement multiple errors in redux side

### LINKED ISSUE #976: Multiple error conditions should be allowable in the redux store
Currently, only a single error condition is storable in the frontend, but it is common (in form validation, for example) to have multiple error conditions.

### PR DESCRIPTION
**Pull request by cellog**
# Description

Fixes #976 

Note that this is a minimal implementation, so the `Error` component simply displays the last error triggered

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/__tests__/actions/error.test.js [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
+import {
+  setError,
+  SET_ERROR,
+  RESET_ERROR,
+  resetError,
+} from '../../actions/error'
 
 describe('error actions', () => {
   it('should create an action to set the error', () => {
-    const error = {
```
Comment (julien51) [inline unlock-app/src/__tests__/actions/error.test.js]: Same comment as in other PR. What is the purpose of this in the context of this PR? 
This will create merge conflicts for you...

File: unlock-app/src/__tests__/reducers/errorReducer.test.js [hunk 2]
```diff
@@ -1,48 +1,45 @@
...
-import reducer from '../../reducers/errorReducer'
-import { SET_ERROR } from '../../actions/error'
+import reducer, { initialState } from '../../reducers/errorReducer'
+import { setError, resetError } from '../../actions/error'
 import { SET_PROVIDER } from '../../actions/provider'
 import { SET_NETWORK } from '../../actions/network'
 
 describe('error reducer', () => {
-  const error = <p>Something was wrong</p>
+  const action = setError({ message: 'hi', context: 'there' })
```
Comment (julien51) [inline unlock-app/src/__tests__/reducers/errorReducer.test.js]: Same comment as on other PR: why doing that in *this* PR? I think this is out of scope.

File: unlock-app/src/actions/error.js [hunk 3]
```diff
@@ -1,6 +1,12 @@
 export const SET_ERROR = 'SET_ERROR'
+export const RESET_ERROR = 'RESET_ERROR'
 
 export const setError = error => ({
   type: SET_ERROR,
   error,
 })
+
+export const resetError = error => ({
```
Comment (julien51) [inline unlock-app/src/actions/error.js]: You only seem to be calling this function with no `error`. Why then have it as an argument?

Comment (julien51) [reply unlock-app/src/actions/error.js]: And then since it seems to remove all error*s* when I would call it __resetError*s*__.

File: unlock-app/src/reducers/errorReducer.js [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
     return initialState
   }
 
   if (action.type === SET_ERROR) {
-    return action.error
+    return [...state, action.error]
+  }
+
+  if (action.type === RESET_ERROR) {
+    if (!action.error) return initialState
```
Comment (julien51) [inline unlock-app/src/reducers/errorReducer.js]: Since you only call this function with no error why the distinction?

**Comment by cellog** [comment]
stripped resetError, rebased and ready for review @julien51

File: unlock-app/src/createUnlockStore.js [hunk 5]
```diff
@@ -53,7 +53,7 @@ export const createUnlockStore = (
     provider: providerReducer,
     transactions: transactionsReducer,
     currency: currencyReducer,
-    error: errorReducer,
+    errors: errorReducer,
```
Comment (julien51) [inline unlock-app/src/createUnlockStore.js]: Stick to `error` since this is `errorReducer` too or switch to `errorsReducer` (my preference since the point of this PR is to add support for multiple errors in the store)

File: unlock-app/src/components/interface/Error.js [hunk 6]
```diff
@@ -23,13 +23,13 @@ export const Error = ({ children, error, close }) => {
   )
 }
 
-const mapStateToProps = ({ error }) => ({
-  error: error,
+const mapStateToProps = ({ errors }) => ({
+  error: errors.length ? errors[errors.length - 1] : null,
```
Comment (julien51) [inline unlock-app/src/components/interface/Error.js]: Maybe a quick comment to indicate that since the UI only supports a single error for now we show the first one?

File: unlock-app/src/createUnlockStore.js (hunk 5 above)
Comment (cellog) [reply unlock-app/src/createUnlockStore.js]: moving to errorsReducer
